docs/track.rst

```diff
@@ -3162,3 +3181,26 @@ You can also specify a number of clients on sub tasks explicitly (by default, on
         }
 
 This will ensure that the phrase query will be executed by two clients. All other ones are executed by one client.
+
+
+.. _track_dependencies:
+
+dependencies
+............
+
+If your track code requires additional Python dependencies at runtime (for instance, for code included in :ref:`custom runners <adding_tracks_custom_runners>` or :ref:`parameter sources <adding_tracks_custom_param_sources>`) you can declare Pip-installable libraries in your track's ``dependencies`` section.
+``dependencies`` is an array of ``pip install`` targets, which Rally will manage when the benchmark begins::
+
+  "dependencies": [
+    "pytoml",
+    "git+https://github.com/elastic/apm-agent-python@main",
+    "eland<8.0.0"
+  ]
+
+
+Rally will manage the installation of the libraries, given the environmentally-available ``pip`` and the target directory ``~/.rally/lib``
+
+.. warning::
+
+    Because modules will be loaded in the same environment as ``esrally`` itself, conflicts with Rally and its own dependencies are possible with the usage of Track Dependencies.
```

Comment:
We need to document that we're using a different directory (maybe say where it is?) and mention that it will have precedence over other environments as we're using `sys.path.insert(0)`.

Reply:
addressed in line 3243 above. the specifics of using `sys.path.insert(0)` feels like implementation detail leak, specifically since the `libs` directory should not be loaded if nothing is declared for dependencies

Reply:
While `sys.path.insert(0)` is an implementation detail, what are your thoughts on just mentioning that the declared `dependencies` will take precedence over already install libraries?